SignaturePipe doesn't initialize bouncycastle

### Current Behavior

When using Signature pipe in a kubernetes pod i'm getting this error during execution of the pipe:

```
FrankElement [org.frankframework.pipes.SignaturePipe@1a6143f2]
org.frankframework.encryption.EncryptionException: unable to open keystore 
[classpath:ProcessBrokerEmailMessage/PrivateKeys/PrivKeyTST.pem]: (KeyStoreException) PEM not found: 
(NoSuchAlgorithmException) PEM KeyStore not available
```

locally this works fine.

I think the issue is that bouncy castle is not loaded during startup and the security provider is not available execution of the pipe run.

i tested this hypothesis by adding a PGP pipe to the configuration which does have this code in the configure block:

Security.addProvider(new BouncyCastleProvider());

and then it works as expected because bouncy castle is loaded in the framework.





### Expected Behavior

signaturepipe should be able to use bouncy castle providers without loading a pgp pipe

### Environment Information

**FF! 9.4.1-b20260508133104**: ibis4felet-main-worker 
Running on **ibis4felet-main-worker-hybrid-c95d589f5-9ls77** using **Apache Tomcat/11.0.18**
Java Version: **OpenJDK Runtime Environment (21.0.10+7-LTS)**
Heap size: **322.5 MiB**, total JVM memory: **614.0 MiB**
Free memory: **291.5 MiB**, max memory: **2.0 GiB**
Free disk space: **21.9 GiB**, total disk space: **29.9 GiB**
Up since: **2026-05-12 16:02:26**, timezone: **Etc/UTC**


### Steps To Reproduce

_No response_

### Configuration

```xml
signature pipe used:


	<SignaturePipe name="CreateHMAC"
			getInputFromFixedValue="Message"
			keystore="ProcessBrokerEmailMessage/PrivateKeys/PrivKeyTST.pem"
			keystoreType="PEM"
			action="SIGN"
			algorithm="SHA512withRSA"
			signatureBase64="true"
			storeResultInSessionKey="hashKeyEncoded">
			<param name="secret" sessionKey="privKey" />
			<forward name="success" path="READY" />
		</SignaturePipe>


pgp used:


		<PGPPipe name="InitBC"
			action="ENCRYPT"
			publicKeys="ProcessBrokerEmailMessage/PrivateKeys/pgp.pem"
			recipients="email@not.set">
			<forward name="success" path="READY" />
		</PGPPipe>
```

### Input

_No response_

### What database are you using?

_No response_

### What browsers are you seeing the problem on?

_No response_

### Relevant Log Output

```console
2026-05-11 14:24:31,754 INFO  [hz.epic_buck.event-5] {} endpoints.SecurityItems - unable to open keystore of FrankElement [org.frankframework.pipes.SignaturePipe@1a6143f2]
org.frankframework.encryption.EncryptionException: unable to open keystore [classpath:ProcessBrokerEmailMessage/PrivateKeys/PrivKeyTST.pem]: (KeyStoreException) PEM not found: (NoSuchAlgorithmException) PEM KeyStore not available
	at org.frankframework.encryption.CorePkiUtil.createKeyStore(CorePkiUtil.java:164) ~[frankframework-core-9.4.1-b20260508133104.jar:9.4.1-b20260508133104]
	at org.frankframework.management.bus.endpoints.SecurityItems.extractKeyStore(SecurityItems.java:313) ~[frankframework-core-9.4.1-b20260508133104.jar:9.4.1-b20260508133104]
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source) ~[?:?]
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source) ~[?:?]
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source) ~[?:?]
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(Unknown Source) ~[?:?]
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(Unknown Source) ~[?:?]
	at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source) ~[?:?]
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source) ~[?:?]
	at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source) ~[?:?]
	at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(Unknown Source) ~[?:?]
	at java.base/java.util.stream.ReferencePipeline.toArray(Unknown Source) ~[?:?]
	at java.base/java.util.stream.ReferencePipeline.toArray(Unknown Source) ~[?:?]
	at java.base/java.util.stream.ReferencePipeline.toList(Unknown Source) ~[?:?]
	at org.frankframework.management.bus.endpoints.SecurityItems.getExpiringCertificates(SecurityItems.java:279) ~[frankframework-core-9.4.1-b20260508133104.jar:9.4.1-b20260508133104]
	at org.frankframework.management.bus.endpoints.SecurityItems.getSecurityItems(SecurityItems.java:95) ~[frankframework-core-9.4.1-b20260508133104.jar:9.4.1-b20260508133104]
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source) ~[?:?]
	... suppressed 4 lines
	at org.springframework.security.authorization.method.AuthorizationManagerBeforeMethodInterceptor.proceed(AuthorizationManagerBeforeMethodInterceptor.java:268) ~[spring-security-core-6.5.7.jar:6.5.7]
	at org.springframework.security.authorization.method.AuthorizationManagerBeforeMethodInterceptor.attemptAuthorization(AuthorizationManagerBeforeMethodInterceptor.java:263) ~[spring-security-core-6.5.7.jar:6.5.7]
	at org.springframework.security.authorization.method.AuthorizationManagerBeforeMethodInterceptor.invoke(AuthorizationManagerBeforeMethodInterceptor.java:196) ~[spring-security-core-6.5.7.jar:6.5.7]
	at org.springframework.security.config.annotation.method.configuration.DeferringMethodInterceptor.invoke(DeferringMethodInterceptor.java:44) ~[spring-security-config-6.5.7.jar:6.5.7]
	...
	at org.springframework.security.authorization.method.AuthorizationManagerBeforeMethodInterceptor.proceed(AuthorizationManagerBeforeMethodInterceptor.java:268) ~[spring-security-core-6.5.7.jar:6.5.7]
	at org.springframework.security.authorization.method.AuthorizationManagerBeforeMethodInterceptor.attemptAuthorization(AuthorizationManagerBeforeMethodInterceptor.java:263) ~[spring-security-core-6.5.7.jar:6.5.7]
	at org.springframework.security.authorization.method.AuthorizationManagerBeforeMethodInterceptor.invoke(AuthorizationManagerBeforeMethodInterceptor.java:196) ~[spring-security-core-6.5.7.jar:6.5.7]
	at org.springframework.security.config.annotation.method.configuration.DeferringMethodInterceptor.invoke(DeferringMethodInterceptor.java:44) ~[spring-security-config-6.5.7.jar:6.5.7]
	... suppressed 2 lines
	at org.frankframework.management.bus.endpoints.SecurityItems$$SpringCGLIB$$0.getSecurityItems(<generated>) ~[frankframework-core-9.4.1-b20260508133104.jar:9.4.1-b20260508133104]
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source) ~[?:?]
	...
	at org.springframework.messaging.handler.invocation.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:169) ~[spring-messaging-6.2.15.jar:6.2.15]
	...
	at org.springframework.messaging.handler.invocation.InvocableHandlerMethod.invoke(InvocableHandlerMethod.java:119) ~[spring-messaging-6.2.15.jar:6.2.15]
	... suppressed 10 lines
	at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:187) ~[spring-messaging-6.2.15.jar:6.2.15]
	at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:166) ~[spring-messaging-6.2.15.jar:6.2.15]
	at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:47) ~[spring-messaging-6.2.15.jar:6.2.15]
	at org.springframework.messaging.core.AbstractMessageSendingTemplate.send(AbstractMessageSendingTemplate.java:109) ~[spring-messaging-6.2.15.jar:6.2.15]
	... suppressed 10 lines
	at org.springframework.integration.dispatcher.BroadcastingDispatcher.invokeHandler(BroadcastingDispatcher.java:235) ~[spring-integration-core-6.5.5.jar:6.5.5]
	at org.springframework.integration.dispatcher.BroadcastingDispatcher.dispatch(BroadcastingDispatcher.java:192) ~[spring-integration-core-6.5.5.jar:6.5.5]
	at org.springframework.integration.channel.AbstractSubscribableChannel.doSend(AbstractSubscribableChannel.java:72) ~[spring-integration-core-6.5.5.jar:6.5.5]
	at org.springframework.integration.channel.AbstractMessageChannel.sendInternal(AbstractMessageChannel.java:436) ~[spring-integration-core-6.5.5.jar:6.5.5]
	at org.springframework.integration.channel.AbstractMessageChannel.send(AbstractMessageChannel.java:343) ~[spring-integration-core-6.5.5.jar:6.5.5]
	at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:187) ~[spring-messaging-6.2.15.jar:6.2.15]
	at org.springframework.messaging.core.GenericMessagingTemplate.doSendAndReceive(GenericMessagingTemplate.java:234) ~[spring-messaging-6.2.15.jar:6.2.15]
	at org.springframework.messaging.core.GenericMessagingTemplate.doSendAndReceive(GenericMessagingTemplate.java:47) ~[spring-messaging-6.2.15.jar:6.2.15]
	at org.springframework.messaging.core.AbstractMessagingTemplate.sendAndReceive(AbstractMessagingTemplate.java:47) ~[spring-messaging-6.2.15.jar:6.2.15]
	at org.springframework.integration.core.MessagingTemplate.sendAndReceive(MessagingTemplate.java:107) ~[spring-integration-core-6.5.5.jar:6.5.5]
	at org.springframework.integration.gateway.MessagingGatewaySupport.doSendAndReceive(MessagingGatewaySupport.java:687) ~[spring-integration-core-6.5.5.jar:6.5.5]
	at org.springframework.integration.gateway.MessagingGatewaySupport.sendAndReceive(MessagingGatewaySupport.java:615) ~[spring-integration-core-6.5.5.jar:6.5.5]
	at org.springframework.integration.gateway.MessagingGatewaySupport.sendAndReceiveMessage(MessagingGatewaySupport.java:588) ~[spring-integration-core-6.5.5.jar:6.5.5]
	at org.frankframework.management.gateway.HazelcastInboundGateway.processMessage(HazelcastInboundGateway.java:137) ~[frankframework-management-gateway-9.4.1-b20260508133104.jar:9.4.1-b20260508133104]
	at org.frankframework.management.gateway.HazelcastInboundGateway.handleIncomingMessage(HazelcastInboundGateway.java:118) ~[frankframework-management-gateway-9.4.1-b20260508133104.jar:9.4.1-b20260508133104]
	at com.hazelcast.topic.impl.TopicService.lambda$dispatchEvent$1(TopicService.java:152) ~[hazelcast-5.5.0.jar:5.5.0]
	at com.hazelcast.internal.namespace.impl.NoOpUserCodeNamespaceService.runWithNamespace(NoOpUserCodeNamespaceService.java:82) ~[hazelcast-5.5.0.jar:5.5.0]
	at com.hazelcast.internal.namespace.NamespaceUtil.runWithNamespace(NamespaceUtil.java:106) ~[hazelcast-5.5.0.jar:5.5.0]
	at com.hazelcast.topic.impl.TopicService.dispatchEvent(TopicService.java:151) ~[hazelcast-5.5.0.jar:5.5.0]
	at com.hazelcast.spi.impl.eventservice.impl.EventProcessor.process(EventProcessor.java:63) ~[hazelcast-5.5.0.jar:5.5.0]
	at com.hazelcast.spi.impl.eventservice.impl.RemoteEventProcessor.run(RemoteEventProcessor.java:48) ~[hazelcast-5.5.0.jar:5.5.0]
	at com.hazelcast.internal.util.executor.StripedExecutor$Worker.process(StripedExecutor.java:244) [hazelcast-5.5.0.jar:5.5.0]
	at com.hazelcast.internal.util.executor.StripedExecutor$Worker.run(StripedExecutor.java:227) [hazelcast-5.5.0.jar:5.5.0]
Caused by: java.security.KeyStoreException: PEM not found
	at java.base/java.security.KeyStore.getInstance(Unknown Source) ~[?:?]
	at org.frankframework.encryption.CommonsPkiUtil.createKeyStore(CommonsPkiUtil.java:85) ~[frankframework-commons-9.4.1-b20260508133104.jar:9.4.1-b20260508133104]
	at org.frankframework.encryption.CorePkiUtil.createKeyStore(CorePkiUtil.java:162) ~[frankframework-core-9.4.1-b20260508133104.jar:9.4.1-b20260508133104]
	... 84 more
Caused by: java.security.NoSuchAlgorithmException: PEM KeyStore not available
	at java.base/sun.security.jca.GetInstance.getInstance(Unknown Source) ~[?:?]
	at java.base/java.security.Security.getImpl(Unknown Source) ~[?:?]
	... 87 more
2026-05-11 14:24:31,757 INFO  [hz.epic_buck.event-5] {} encryption.CommonsPkiUtil - Initializing keystore from url [classpath:ProcessBrokerEmailMessage/PrivateKeys/PrivKeyPRD.pem]
2026-05-11 14:24:31,757 INFO  [hz.epic_buck.event-5] {} endpoints.SecurityItems - unable to open keystore of FrankElement [org.frankframework.pipes.SignaturePipe@7039da63]
org.frankframework.encryption.EncryptionException: unable to open keystore [classpath:ProcessBrokerEmailMessage/PrivateKeys/PrivKeyPRD.pem]: (KeyStoreException) PEM not found: (NoSuchAlgorithmException) PEM KeyStore not available
	at org.frankframework.encryption.CorePkiUtil.createKeyStore(CorePkiUtil.java:164) ~[frankframework-core-9.4.1-b20260508133104.jar:9.4.1-b20260508133104]
	at org.frankframework.management.bus.endpoints.SecurityItems.extractKeyStore(SecurityItems.java:313) ~[frankframework-core-9.4.1-b20260508133104.jar:9.4.1-b20260508133104]
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source) ~[?:?]
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source) ~[?:?]
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source) ~[?:?]
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(Unknown Source) ~[?:?]
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(Unknown Source) ~[?:?]
	at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source) ~[?:?]
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source) ~[?:?]
	at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source) ~[?:?]
	at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(Unknown Source) ~[?:?]
	at java.base/java.util.stream.ReferencePipeline.toArray(Unknown Source) ~[?:?]
	at java.base/java.util.stream.ReferencePipeline.toArray(Unknown Source) ~[?:?]
	at java.base/java.util.stream.ReferencePipeline.toList(Unknown Source) ~[?:?]
	at org.frankframework.management.bus.endpoints.SecurityItems.getExpiringCertificates(SecurityItems.java:279) ~[frankframework-core-9.4.1-b20260508133104.jar:9.4.1-b20260508133104]
	at org.frankframework.management.bus.endpoints.SecurityItems.getSecurityItems(SecurityItems.java:95) ~[frankframework-core-9.4.1-b20260508133104.jar:9.4.1-b20260508133104]
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source) ~[?:?]
	... suppressed 4 lines
	at org.springframework.security.authorization.method.AuthorizationManagerBeforeMethodInterceptor.proceed(AuthorizationManagerBeforeMethodInterceptor.java:268) ~[spring-security-core-6.5.7.jar:6.5.7]
	at org.springframework.security.authorization.method.AuthorizationManagerBeforeMethodInterceptor.attemptAuthorization(AuthorizationManagerBeforeMethodInterceptor.java:263) ~[spring-security-core-6.5.7.jar:6.5.7]
	at org.springframework.security.authorization.method.AuthorizationManagerBeforeMethodInterceptor.invoke(AuthorizationManagerBeforeMethodInterceptor.java:196) ~[spring-security-core-6.5.7.jar:6.5.7]
	at org.springframework.security.config.annotation.method.configuration.DeferringMethodInterceptor.invoke(DeferringMethodInterceptor.java:44) ~[spring-security-config-6.5.7.jar:6.5.7]
	...
	at org.springframework.security.authorization.method.AuthorizationManagerBeforeMethodInterceptor.proceed(AuthorizationManagerBeforeMethodInterceptor.java:268) ~[spring-security-core-6.5.7.jar:6.5.7]
	at org.springframework.security.authorization.method.AuthorizationManagerBeforeMethodInterceptor.attemptAuthorization(AuthorizationManagerBeforeMethodInterceptor.java:263) ~[spring-security-core-6.5.7.jar:6.5.7]
	at org.springframework.security.authorization.method.AuthorizationManagerBeforeMethodInterceptor.invoke(AuthorizationManagerBeforeMethodInterceptor.java:196) ~[spring-security-core-6.5.7.jar:6.5.7]
	at org.springframework.security.config.annotation.method.configuration.DeferringMethodInterceptor.invoke(DeferringMethodInterceptor.java:44) ~[spring-security-config-6.5.7.jar:6.5.7]
	... suppressed 2 lines
	at org.frankframework.management.bus.endpoints.SecurityItems$$SpringCGLIB$$0.getSecurityItems(<generated>) ~[frankframework-core-9.4.1-b20260508133104.jar:9.4.1-b20260508133104]
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source) ~[?:?]
	...
	at org.springframework.messaging.handler.invocation.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:169) ~[spring-messaging-6.2.15.jar:6.2.15]
	...
	at org.springframework.messaging.handler.invocation.InvocableHandlerMethod.invoke(InvocableHandlerMethod.java:119) ~[spring-messaging-6.2.15.jar:6.2.15]
	... suppressed 10 lines
	at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:187) ~[spring-messaging-6.2.15.jar:6.2.15]
	at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:166) ~[spring-messaging-6.2.15.jar:6.2.15]
	at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:47) ~[spring-messaging-6.2.15.jar:6.2.15]
	at org.springframework.messaging.core.AbstractMessageSendingTemplate.send(AbstractMessageSendingTemplate.java:109) ~[spring-messaging-6.2.15.jar:6.2.15]
	... suppressed 10 lines
	at org.springframework.integration.dispatcher.BroadcastingDispatcher.invokeHandler(BroadcastingDispatcher.java:235) ~[spring-integration-core-6.5.5.jar:6.5.5]
	at org.springframework.integration.dispatcher.BroadcastingDispatcher.dispatch(BroadcastingDispatcher.java:192) ~[spring-integration-core-6.5.5.jar:6.5.5]
	at org.springframework.integration.channel.AbstractSubscribableChannel.doSend(AbstractSubscribableChannel.java:72) ~[spring-integration-core-6.5.5.jar:6.5.5]
	at org.springframework.integration.channel.AbstractMessageChannel.sendInternal(AbstractMessageChannel.java:436) ~[spring-integration-core-6.5.5.jar:6.5.5]
	at org.springframework.integration.channel.AbstractMessageChannel.send(AbstractMessageChannel.java:343) ~[spring-integration-core-6.5.5.jar:6.5.5]
	at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:187) ~[spring-messaging-6.2.15.jar:6.2.15]
	at org.springframework.messaging.core.GenericMessagingTemplate.doSendAndReceive(GenericMessagingTemplate.java:234) ~[spring-messaging-6.2.15.jar:6.2.15]
	at org.springframework.messaging.core.GenericMessagingTemplate.doSendAndReceive(GenericMessagingTemplate.java:47) ~[spring-messaging-6.2.15.jar:6.2.15]
	at org.springframework.messaging.core.AbstractMessagingTemplate.sendAndReceive(AbstractMessagingTemplate.java:47) ~[spring-messaging-6.2.15.jar:6.2.15]
	at org.springframework.integration.core.MessagingTemplate.sendAndReceive(MessagingTemplate.java:107) ~[spring-integration-core-6.5.5.jar:6.5.5]
	at org.springframework.integration.gateway.MessagingGatewaySupport.doSendAndReceive(MessagingGatewaySupport.java:687) ~[spring-integration-core-6.5.5.jar:6.5.5]
	at org.springframework.integration.gateway.MessagingGatewaySupport.sendAndReceive(MessagingGatewaySupport.java:615) ~[spring-integration-core-6.5.5.jar:6.5.5]
	at org.springframework.integration.gateway.MessagingGatewaySupport.sendAndReceiveMessage(MessagingGatewaySupport.java:588) ~[spring-integration-core-6.5.5.jar:6.5.5]
	at org.frankframework.management.gateway.HazelcastInboundGateway.processMessage(HazelcastInboundGateway.java:137) ~[frankframework-management-gateway-9.4.1-b20260508133104.jar:9.4.1-b20260508133104]
	at org.frankframework.management.gateway.HazelcastInboundGateway.handleIncomingMessage(HazelcastInboundGateway.java:118) ~[frankframework-management-gateway-9.4.1-b20260508133104.jar:9.4.1-b20260508133104]
	at com.hazelcast.topic.impl.TopicService.lambda$dispatchEvent$1(TopicService.java:152) ~[hazelcast-5.5.0.jar:5.5.0]
	at com.hazelcast.internal.namespace.impl.NoOpUserCodeNamespaceService.runWithNamespace(NoOpUserCodeNamespaceService.java:82) ~[hazelcast-5.5.0.jar:5.5.0]
	at com.hazelcast.internal.namespace.NamespaceUtil.runWithNamespace(NamespaceUtil.java:106) ~[hazelcast-5.5.0.jar:5.5.0]
	at com.hazelcast.topic.impl.TopicService.dispatchEvent(TopicService.java:151) ~[hazelcast-5.5.0.jar:5.5.0]
	at com.hazelcast.spi.impl.eventservice.impl.EventProcessor.process(EventProcessor.java:63) ~[hazelcast-5.5.0.jar:5.5.0]
	at com.hazelcast.spi.impl.eventservice.impl.RemoteEventProcessor.run(RemoteEventProcessor.java:48) ~[hazelcast-5.5.0.jar:5.5.0]
	at com.hazelcast.internal.util.executor.StripedExecutor$Worker.process(StripedExecutor.java:244) [hazelcast-5.5.0.jar:5.5.0]
	at com.hazelcast.internal.util.executor.StripedExecutor$Worker.run(StripedExecutor.java:227) [hazelcast-5.5.0.jar:5.5.0]
Caused by: java.security.KeyStoreException: PEM not found
	at java.base/java.security.KeyStore.getInstance(Unknown Source) ~[?:?]
	at org.frankframework.encryption.CommonsPkiUtil.createKeyStore(CommonsPkiUtil.java:85) ~[frankframework-commons-9.4.1-b20260508133104.jar:9.4.1-b20260508133104]
	at org.frankframework.encryption.CorePkiUtil.createKeyStore(CorePkiUtil.java:162) ~[frankframework-core-9.4.1-b20260508133104.jar:9.4.1-b20260508133104]
	... 84 more
Caused by: java.security.NoSuchAlgorithmException: PEM KeyStore not available
	at java.base/sun.security.jca.GetInstance.getInstance(Unknown Source) ~[?:?]
	at java.base/java.security.Security.getImpl(Unknown Source) ~[?:?]
	... 87 more
```

### Anything else?

_No response_

--- a/core/src/main/java/org/frankframework/pipes/PGPPipe.java
+++ b/core/src/main/java/org/frankframework/pipes/PGPPipe.java
@@ -98,7 +98,7 @@ public class PGPPipe extends FixedForwardPipe {
 		if (action == null) {
 			throw new ConfigurationException("Action can not be null!");
 		}
-		if(Security.getProvider(BouncyCastleProvider.PROVIDER_NAME) != null) {
+		if (Security.getProvider(BouncyCastleProvider.PROVIDER_NAME) != null) {
 			Security.removeProvider(BouncyCastleProvider.PROVIDER_NAME);
 		}
 

--- a/core/src/main/java/org/frankframework/pipes/SignaturePipe.java
+++ b/core/src/main/java/org/frankframework/pipes/SignaturePipe.java
@@ -28,6 +28,7 @@ import java.security.SignatureException;
 
 import org.apache.commons.codec.binary.Base64;
 import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.Strings;
 import org.bouncycastle.jce.provider.BouncyCastleProvider;
 import org.jspecify.annotations.NonNull;
 
@@ -97,13 +98,15 @@ public class SignaturePipe extends FixedForwardPipe implements HasKeystore {
 		if (getKeystoreType() == KeystoreType.PEM && Security.getProvider(BouncyCastleProvider.PROVIDER_NAME) == null) {
 			Security.addProvider(new BouncyCastleProvider());
 		}
-		if (!Security.getAlgorithms("Signature").contains(getAlgorithm())  && Security.getProvider(BouncyCastleProvider.PROVIDER_NAME) == null) {
+
+		String signingAlgorithm = getAlgorithm();
+		if (!isSigningAlgorithmAvailable(signingAlgorithm) && Security.getProvider(BouncyCastleProvider.PROVIDER_NAME) == null) {
 			// If the algorithm is not supported, first try if that is solved by adding the BouncyCastle provider
 			Security.addProvider(new BouncyCastleProvider());
 		}
 
-		if (!Security.getAlgorithms("Signature").contains(getAlgorithm())) {
-			throw new ConfigurationException("Signature algorithm [" + getAlgorithm() + "] not supported, supported algorithms: " + Security.getAlgorithms("Signature"));
+		if (!isSigningAlgorithmAvailable(signingAlgorithm)) {
+			throw new ConfigurationException("Signature algorithm [" + signingAlgorithm + "] not supported, supported algorithms: " + Security.getAlgorithms("Signature"));
 		}
 
 		AuthSSLContextFactory.verifyKeystoreConfiguration(this, null);
@@ -116,6 +119,10 @@ public class SignaturePipe extends FixedForwardPipe implements HasKeystore {
 				throw new ConfigurationException("Forward [failure] must be specified for action [" + action + "]");
 			}
 		}
+	}
+
+	private static boolean isSigningAlgorithmAvailable(String signingAlgorithm) {
+		return Strings.CI.equalsAny(signingAlgorithm, Security.getAlgorithms("Signature").toArray(String[]::new));
 	}
 
 	@Override

--- a/core/src/main/java/org/frankframework/pipes/SignaturePipe.java
+++ b/core/src/main/java/org/frankframework/pipes/SignaturePipe.java
@@ -95,16 +95,11 @@ public class SignaturePipe extends FixedForwardPipe implements HasKeystore {
 		if (StringUtils.isEmpty(this.getKeystore())) {
 			throw new ConfigurationException("keystore must be specified");
 		}
-		if (getKeystoreType() == KeystoreType.PEM && isNoBouncyCastleLoaded()) {
-			Security.addProvider(new BouncyCastleProvider());
-		}
-
 		String signingAlgorithm = getAlgorithm();
-		if (isSigningAlgorithmMissing(signingAlgorithm) && isNoBouncyCastleLoaded()) {
-			// If the algorithm is not supported, first try if that is solved by adding the BouncyCastle provider
+		if (isNoBouncyCastleLoaded() && (getKeystoreType() == KeystoreType.PEM || isSigningAlgorithmMissing(signingAlgorithm))) {
 			Security.addProvider(new BouncyCastleProvider());
 		}
-
+		// Adding BouncyCastle may add the missing algorithm, check again after loading BC.
 		if (isSigningAlgorithmMissing(signingAlgorithm)) {
 			throw new ConfigurationException("Signature algorithm [" + signingAlgorithm + "] not supported, supported algorithms: " + Security.getAlgorithms("Signature"));
 		}

--- a/core/src/main/java/org/frankframework/pipes/SignaturePipe.java
+++ b/core/src/main/java/org/frankframework/pipes/SignaturePipe.java
@@ -95,34 +95,38 @@ public class SignaturePipe extends FixedForwardPipe implements HasKeystore {
 		if (StringUtils.isEmpty(this.getKeystore())) {
 			throw new ConfigurationException("keystore must be specified");
 		}
-		if (getKeystoreType() == KeystoreType.PEM && Security.getProvider(BouncyCastleProvider.PROVIDER_NAME) == null) {
+		if (getKeystoreType() == KeystoreType.PEM && isNoBouncyCastleLoaded()) {
 			Security.addProvider(new BouncyCastleProvider());
 		}
 
 		String signingAlgorithm = getAlgorithm();
-		if (!isSigningAlgorithmAvailable(signingAlgorithm) && Security.getProvider(BouncyCastleProvider.PROVIDER_NAME) == null) {
+		if (isSigningAlgorithmMissing(signingAlgorithm) && isNoBouncyCastleLoaded()) {
 			// If the algorithm is not supported, first try if that is solved by adding the BouncyCastle provider
 			Security.addProvider(new BouncyCastleProvider());
 		}
 
-		if (!isSigningAlgorithmAvailable(signingAlgorithm)) {
+		if (isSigningAlgorithmMissing(signingAlgorithm)) {
 			throw new ConfigurationException("Signature algorithm [" + signingAlgorithm + "] not supported, supported algorithms: " + Security.getAlgorithms("Signature"));
 		}
 
 		AuthSSLContextFactory.verifyKeystoreConfiguration(this, null);
 		if (getAction() == Action.VERIFY) {
 			if (!getParameterList().hasParameter(PARAMETER_SIGNATURE)) {
-				throw new ConfigurationException("Parameter [" + PARAMETER_SIGNATURE + "] must be specfied for action [" + action + "]");
+				throw new ConfigurationException("Parameter [" + PARAMETER_SIGNATURE + "] must be specified for action [" + action + "]");
 			}
 			failureForward = findForward("failure");
-			if (failureForward==null)  {
+			if (failureForward == null)  {
 				throw new ConfigurationException("Forward [failure] must be specified for action [" + action + "]");
 			}
 		}
 	}
 
-	private static boolean isSigningAlgorithmAvailable(String signingAlgorithm) {
-		return Strings.CI.equalsAny(signingAlgorithm, Security.getAlgorithms("Signature").toArray(String[]::new));
+	private static boolean isNoBouncyCastleLoaded() {
+		return Security.getProvider(BouncyCastleProvider.PROVIDER_NAME) == null;
+	}
+
+	private static boolean isSigningAlgorithmMissing(String signingAlgorithm) {
+		return !Strings.CI.equalsAny(signingAlgorithm, Security.getAlgorithms("Signature").toArray(String[]::new));
 	}
 
 	@Override

--- a/core/src/main/java/org/frankframework/pipes/SignaturePipe.java
+++ b/core/src/main/java/org/frankframework/pipes/SignaturePipe.java
@@ -22,11 +22,13 @@ import java.security.NoSuchAlgorithmException;
 import java.security.NoSuchProviderException;
 import java.security.PrivateKey;
 import java.security.PublicKey;
+import java.security.Security;
 import java.security.Signature;
 import java.security.SignatureException;
 
 import org.apache.commons.codec.binary.Base64;
 import org.apache.commons.lang3.StringUtils;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
 import org.jspecify.annotations.NonNull;
 
 import lombok.Getter;
@@ -44,6 +46,7 @@ import org.frankframework.encryption.CorePkiUtil;
 import org.frankframework.encryption.EncryptionException;
 import org.frankframework.encryption.HasKeystore;
 import org.frankframework.encryption.KeystoreConfiguration;
+import org.frankframework.encryption.KeystoreType;
 import org.frankframework.lifecycle.LifecycleException;
 import org.frankframework.parameters.ParameterValueList;
 import org.frankframework.stream.Message;
@@ -90,6 +93,17 @@ public class SignaturePipe extends FixedForwardPipe implements HasKeystore {
 		}
 		if (StringUtils.isEmpty(this.getKeystore())) {
 			throw new ConfigurationException("keystore must be specified");
+		}
+		if (getKeystoreType() == KeystoreType.PEM && Security.getProvider(BouncyCastleProvider.PROVIDER_NAME) == null) {
+			Security.addProvider(new BouncyCastleProvider());
+		}
+		if (!Security.getAlgorithms("Signature").contains(getAlgorithm())  && Security.getProvider(BouncyCastleProvider.PROVIDER_NAME) == null) {
+			// If the algorithm is not supported, first try if that is solved by adding the BouncyCastle provider
+			Security.addProvider(new BouncyCastleProvider());
+		}
+
+		if (!Security.getAlgorithms("Signature").contains(getAlgorithm())) {
+			throw new ConfigurationException("Signature algorithm [" + getAlgorithm() + "] not supported, supported algorithms: " + Security.getAlgorithms("Signature"));
 		}
 
 		AuthSSLContextFactory.verifyKeystoreConfiguration(this, null);

--- a/core/src/test/java/org/frankframework/pipes/SignaturePipeTest.java
+++ b/core/src/test/java/org/frankframework/pipes/SignaturePipeTest.java
@@ -11,14 +11,17 @@ import static org.junit.jupiter.api.Assertions.fail;
 import java.net.URL;
 import java.security.KeyStore;
 import java.security.PrivateKey;
+import java.security.Security;
 import java.util.Arrays;
 
 import javax.net.ssl.KeyManager;
 import javax.net.ssl.X509KeyManager;
 
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
 import org.hamcrest.Matchers;
 import org.junit.jupiter.api.Test;
 
+import org.frankframework.configuration.ConfigurationException;
 import org.frankframework.core.PipeForward;
 import org.frankframework.core.PipeRunResult;
 import org.frankframework.encryption.CommonsPkiUtil;
@@ -69,7 +72,7 @@ public class SignaturePipeTest extends PipeTestBase<SignaturePipe> {
 					alias = aliases[0];
 				}
 			} catch (Exception e) {
-				fail("unable to retreive alias from PFX file");
+				fail("unable to retrieve alias from PFX file");
 			}
 		}
 		assertNotNull(privateKey, aliases != null ? ("found aliases "+Arrays.asList(aliases)+" in PFX file") : "no aliases found in PFX file");
@@ -202,6 +205,7 @@ public class SignaturePipeTest extends PipeTestBase<SignaturePipe> {
 
 	@Test
 	void testVerifyOKPEM() throws Exception {
+		// Arrange
 		pipe.setAction(Action.VERIFY);
 		pipe.setKeystore("/Signature/certificate.crt");
 		pipe.setKeystoreType(KeystoreType.PEM);
@@ -211,12 +215,36 @@ public class SignaturePipeTest extends PipeTestBase<SignaturePipe> {
 		PipeForward failure = new PipeForward();
 		failure.setName("failure");
 		pipe.addForward(failure);
+
+		Security.removeProvider(BouncyCastleProvider.PROVIDER_NAME);
+
 		configureAndStartPipe();
 
+		// Act
 		PipeRunResult prr = doPipe(new Message(testMessage));
 
+		// Assert
 		assertEquals(testMessage, prr.getResult().asString());
 		assertEquals("success", prr.getPipeForward().getName());
+		assertNotNull(Security.getProvider(BouncyCastleProvider.PROVIDER_NAME));
 	}
 
+	@Test
+	void testAlgorithmNotFound() {
+		// Arrange
+		pipe.setAction(Action.VERIFY);
+		pipe.setAlgorithm("bad_algorithm");
+		pipe.setKeystore("/Signature/ks_multipassword.jks");
+		pipe.setKeystorePassword("geheim");
+		pipe.setKeystoreType(KeystoreType.JKS);
+
+		Security.removeProvider(BouncyCastleProvider.PROVIDER_NAME);
+
+		// Act
+		ConfigurationException ce = assertThrows(ConfigurationException.class, pipe::configure);
+
+		// Assert
+		assertThat(ce.getMessage(), Matchers.startsWith("Signature algorithm [bad_algorithm] not supported"));
+		assertNotNull(Security.getProvider(BouncyCastleProvider.PROVIDER_NAME));
+	}
 }

